### ISSUE TITLE
 A11y_AzureDataStudio_ SaveJupyterBookPreviewTab_HighContrast : Color contrast ratio fails for the text on the link .

### ISSUE THREAD

**Issue by 12-shweta** [microsoft/azuredatastudio#7793]
[Check out Accessibility Insights!](https://accessibilityinsights.io/)- Identify accessibility bugs before check-in and make bug fixing faster and easier.”

GitHubTags : #A11yTCS;#DesktopApp;#electron.0;#HighContrast;#WCAG1.4.3;#A11y_AzureDataStudio;#SQL Azure Data Studio;#Win32;#A11yMAS

**Environment Details:	**
Application Name: Azure Data Studio
Version: 1.12.0-insider (user setup)
Commit:6582debd7388b1be512fc3a9b956d3e54f6115e7
Date: 2019-10-03T08:56:27.454Z
VS Code: 1.38.0
Electron: 4.2.10
Chrome: 69.0.3497.128
Node.js: 10.11.0
V8: 6.9.427.31-electron.0
OS: Windows_NT x64 10.0.18362

**Prerequisites :**
Apply High Contrast Black and White Mode.

**Additional Details:**
Tool Name : ColorContrast

**Steps to Reproduce:**
1.Install and Launch the application the azure data studio application.
2.Connect to the servers with the required credentials.
3.Navigate to the control panel and click on the "jupyter books" icon.
4.Press CRTL+SHIFT+P and search "jupyter books: SQL Server 2019 Guide" 
5.Navigate to the left section of jupyter books under the untitled books and click on "save" icon .
6.Navigate to the saved books and click on to the "Home". 
7.Apply high contrast black and white theme .
8.Open AI4D and check the color contrast ratio for the link.

**Actual:**
Color contrast ratio for the text present on the link is less than 7:1.
For example , for “Monitor kubernetes” the color contrast ratio is 6.842:1 .

*Expected:** 
Color contrast ratio should be 7:1 for all the text .

**User Impact:**
User with low vision will find it difficult to see the text present on the screen.

**Attachment for reference :**
![7793_A11y_AzureDataStudio_ SaveJupyterBookPreviewTab_HighContrast](https://user-images.githubusercontent.com/54656883/67071474-4703bf00-f1a0-11e9-81ab-18e3eaf01342.png)

Does this issue occur when all extensions are disabled?: Yes

**Comment by chlafreniere** [comment]
@12-shweta can you help me understand where the 7:1 contrast ratio requirement is defined? I was under the impression that 4.5:1 contrast ratio was sufficient. Has there been a change?

**Comment by 12-shweta** [comment]
**@chlafreniere** In normal contrast the color contrast ratio should be 4.5:1 but if we applied high contrast then ratio should be 7:1 .

**Comment by chlafreniere** [comment]
This is a vs code color that's set in src/vs/platform/theme/common/colorRegistry.ts: https://github.com/microsoft/vscode/blob/d3cda21f0e7e3187d53b4db9c52465f1cc660d74/src/vs/platform/theme/common/colorRegistry.ts#L194-L195.

**Comment by chlafreniere** [comment]
There has been a lot of discussion around this bug internally, so here's a high-level summary:
- The accessibility team internally agree that there does not appear to be a MAS standard for 7:1 beyond the common requirement of 4.5:1:

"
![image](https://user-images.githubusercontent.com/40371649/71224901-a313be00-228b-11ea-848c-673eb7b7819a.png)
"

- The discussion then turned to whether there could be any MAS violations in Azure Data Studio related to this bug (which shares much of its codebase with VS Code, including in this case, with default theme fallback colors). MAS 4.3.1 states that apps will not disrupt accessibility features that the customer has turned on for the platform. For Windows, high contrast mode can be customized at the OS level, and the defaults there far surpass 7:1 (and generally surpass even 14:1). Since Azure Data Studio (and VS Code) do not currently respect these settings, we have therefore filed https://github.com/microsoft/vscode/issues/87298. If the VS Code team decides to fix that particular issue, then we will also have the same functionality in Azure Data Studio.

For these reasons, closing this particular bug.

**Comment by 12-shweta** [comment]
@chlafreniere 
In [MAS1.4.3](https://microsoft.sharepoint.com/:w:/r/teams/msenable/_layouts/15/WopiFrame.aspx?sourcedoc={a73546c7-765f-489c-b18f-afb659fe99e6}) .
We have to check the Color contrast ratio as per the follows:
Normal Contrast mode it is 4.5:1
High contrast it is 7:1.

We need to Check both normal and high contrast mode for all the text present on the screen.
This issue mainly targets especially for low vision users as they work only after applying High contrast Black/White themes .

I am attaching the screenshot taken form the this link [https://www.w3.org/WAI/WCAG21/Understanding/contrast-minimum.html#intent](url)

![image](https://user-images.githubusercontent.com/54656883/72138882-bf4eda80-33b3-11ea-9826-b436910a81e9.png)
